bamboo/unit_tests/test_unit_layer_leaky_relu_distconv.py

```
@@ -0,0 +1,206 @@
+    # LBANN implementation
+    x = x_lbann
+    x = lbann.Reshape(x, dims="4 2 6")
+    y = lbann.LeakyRelu(x, negative_slope=0.01,
+                        data_layout='data_parallel',
+                        parallel_strategy=create_parallel_strategy(4))
+    y = lbann.Reshape(y, dims=str(sample_dims()))    
+    z = lbann.L2Norm2(y)
+    obj.append(z)
+    metrics.append(lbann.Metric(z, name='data-parallel layout'))
+
+    # NumPy implementation
+    vals = []
+    for i in range(num_samples()):
+        x = get_sample(i).astype(np.float64)
+        y = np.where(x > 0, x, 0.01*x)
+        z = tools.numpy_l2norm2(y)
+        vals.append(z)
+    val = np.mean(vals)
+    tol = 8 * val * np.finfo(np.float32).eps
+    callbacks.append(lbann.CallbackCheckMetric(
+        metric=metrics[-1].name,
+        lower_bound=val-tol,
+        upper_bound=val+tol,
+        error_on_failure=True,
+        execution_modes='test'))
+
+    # ------------------------------------------
+    # Model-parallel layout
+    # ------------------------------------------
+
+    # LBANN implementation
+    x = x_lbann
+    x = lbann.Reshape(x, dims="4 2 6")    
+    y = lbann.LeakyRelu(x, negative_slope=2,
+                        data_layout='model_parallel',
+                        parallel_strategy=create_parallel_strategy(4))
+    y = lbann.Reshape(y, dims=str(sample_dims()))    
+    z = lbann.L2Norm2(y)
+    obj.append(z)
+    metrics.append(lbann.Metric(z, name='model-parallel layout'))
+
+    # NumPy implementation
+    vals = []
+    for i in range(num_samples()):
+        x = get_sample(i).astype(np.float64)
+        y = np.where(x > 0, x, 2*x)
+        z = tools.numpy_l2norm2(y)
+        vals.append(z)
+    val = np.mean(vals)
+    tol = 8 * val * np.finfo(np.float32).eps
+    callbacks.append(lbann.CallbackCheckMetric(
+        metric=metrics[-1].name,
+        lower_bound=val-tol,
+        upper_bound=val+tol,
+        error_on_failure=True,
+        execution_modes='test'))
+
```

This part of the test doesn't involve distconv and can be deleted.
```suggestion
```

---

It does not because `model_parallel` is not supported, but it makes sure having a parallel strategy does not induce any error whether the layout is `data_parallel` or `model_parallel`.

---

I would say it's an invalid configuration, so we should throw an exception.

---

While I agree that's a reasonable behavior, the overall policy so far is to disable distconv and fall back to LBANN, rather than throwing exceptions. I added a note to #1470. 

In any case, the test should still be here and make sure the corresponding exception is thrown.